gluten-data/src/main/java/org/apache/gluten/vectorized/NativeColumnarToRowJniWrapper.java

```diff
@@ -38,8 +38,8 @@ public long handle() {
 
   public native long nativeColumnarToRowInit(long memoryManagerHandle) throws RuntimeException;
 
-  public native NativeColumnarToRowInfo nativeColumnarToRowConvert(long batchHandle, long c2rHandle)
-      throws RuntimeException;
+  public native NativeColumnarToRowInfo nativeColumnarToRowConvert(
+      long batchHandle, long c2rHandle, long rowId) throws RuntimeException;
```

Comment:
Can we pass memoryThreshold to jni ? It should be easy to track the config at java side.